src/openrct2/scenario/scenario.h

```diff
@@ -132,8 +132,14 @@ typedef struct rct_s6_data {
 	uint8 pad_01357400[4];
 	uint32 researched_ride_types[8];
 	uint32 researched_ride_entries[8];
-	uint32 researched_track_types_a[128];
-	uint32 researched_track_types_b[128];
+	union {
+		struct {
+			uint32 researched_track_types_a[128];
+			uint32 researched_track_types_b[128];
+		};
+		uint64 researched_track_types[128];
```

Comment:
The other place only combines two `uint32`s. You're trying to combine two arrays.